app/actions/handlers.py

```diff
@@ -0,0 +1,179 @@
+import datetime
+import httpx
+import logging
+import stamina
+import app.actions.client as client
+
+from app.actions.configurations import AuthenticateConfig, FetchSamplesConfig, PullObservationsConfig
+from app.services.activity_logger import activity_logger
+from app.services.gundi import send_observations_to_gundi
+from app.services.state import IntegrationStateManager
+
+
+logger = logging.getLogger(__name__)
+
+
+state_manager = IntegrationStateManager()
+
+
+async def filter_and_transform(devices, integration_id, action_id):
+    def transform(device):
+        device_id = device.pop("Imeino")
+        device_name = device.pop("Vehicle_Name")
```

Comment:
ant reason to use `pop()` instead of `get() ` here? My concern is that with pop you are removing data from the device, so this function will have a side effect on the data it receives as input.

Reply:
I see, I added a suggestion related to that when building the additional dict